staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -216,6 +259,36 @@ func (s *Serializer) Decode(originalData []byte, gvk *schema.GroupVersionKind, i
 	if err := caseSensitiveJsonIterator.Unmarshal(data, obj); err != nil {
 		return nil, actual, err
 	}
+
+	// If the deserializer is non-strict, return successfully here.
+	if !s.strict {
+		return obj, actual, nil
+	}
+
+	// In strict mode pass the data trough the YAMLToJSONStrict converter.
+	// This is done to catch duplicate fields regardless of encoding (JSON or YAML). For JSON data,
+	// the output would equal the input, unless there is a parsing error such as duplicate fields.
+	// As we know this was successful in the non-strict case, the only error that may be returned here
+	// is because of the newly-added strictness; hence we know we can return the typed DuplicateFieldError.
+	altered, err := yaml.YAMLToJSONStrict(originalData)
+	if err != nil {
+		yamlToJSONError := &runtime.DuplicateFieldError{
+			StrictDecoderError: *runtime.NewStrictDecoderError(err.Error(), *actual, originalData),
+		}
+		return nil, actual, yamlToJSONError
+	}
+	// As performance is not an issue for now for the strict deserializer (one have regardless to do
+	// the unmarshal twice); we take the sanitized, altered data that is guaranteed to have no duplicated
+	// fields, and unmarshal this into a copy of the already-populated obj. Any error that occurs here is
+	// due to that a matching field doesn't exist in the object; hence we can return a typed UnknownFieldError.
+	strictObj := obj.DeepCopyObject()
+	if err := strictCaseSensitiveJsonIterator.Unmarshal(altered, strictObj); err != nil {
```

Comment:
the deep copy might have filled pointers with value. This is not an empty struct of the same type as `obj`. I thinking that the json iter unmarshalers always expect a zero-value of the target type.

Reply:
yep, I was unsure about this as well. We might wanna use just `scheme.New()`

Reply:
> This is not an empty struct of the same type as obj

I expected us to copy obj before decoding into it. That should result in an object that is deepequal to the original obj, right?